CMakeLists.txt

```diff
@@ -24,9 +24,24 @@ option(
     ${PROJECT_IS_TOP_LEVEL}
 )
 
-include(CTest)
+add_library(beman.exemplar INTERFACE)
+add_library(beman::exemplar ALIAS beman.exemplar)
+
+target_sources(
+    beman.exemplar
+    PUBLIC
+        FILE_SET HEADERS
+            BASE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/include"
+            FILES
+                "${CMAKE_CURRENT_SOURCE_DIR}/include/beman/exemplar/identity.hpp"
```

Comment:
It's interesting how the header list moves to the top-level directory when it's a header-only library. This has two drawbacks:

1. A header-only -> with-source-file transition is more dramatic.
2. For a repository with multiple libraries, the top-level CMake file would grow quite large and we lose the "one CMakeLists.txt file per library" separation of concerns.

What are your thoughts on peeing the `src/beman/exemplar` directory and having it contain only a `CMakeLists.txt` that builds the target?